tests/snmp/test_snmp_interfaces.py

```diff
@@ -7,22 +7,23 @@
 ]
 
 
-def collect_all_facts(duthost):
+def collect_all_facts(duthost, ports_list, namespace=None):
     """
     Collect all data needed for test per each port from DUT
     :param duthost: DUT host object
     :return: dict with data collected from DUT per each port
     """
     result = {}
-    setup = duthost.setup()['ansible_facts']
-    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
-    cmd = 'redis-cli -n 0 --raw hget "PORT_TABLE:{}" "{}"'
-    ports_list = []
-    _ = [ports_list.extend(config_facts.get(i, {}).keys())
-         for i in ['port_name_to_alias_map', 'PORTCHANNEL', 'MGMT_INTERFACE']]
+    setup = duthost.interface_facts(namespace=namespace)['ansible_facts']['ansible_interface_facts']
+    config_facts = duthost.config_facts(host=duthost.hostname, source="running", namespace=namespace)['ansible_facts']
+
+    if namespace is None or namespace == "":
```

Comment:
>namespace is None or namespace == "" [](http://example.com/codeflow?start=7&length=36)

could be simplified as just `if not namespace:`

Reply:
Updated.